zabbix-monitoring-cluster/nested/clusterNodes.json

```diff
@@ -0,0 +1,206 @@
+    "storageAccountName": "[concat(variables('prefix'),'sto')]",
+    "vmStorageAccountContainerName": "vhds",
+    "vmName": "[concat(variables('prefix'),'vm')]",
+    "vmSize": "Standard_D2",
+    "storageAccountType": "Standard_LRS",
+    "VMCount": "[parameters('vmCount')]",
+    "VMStart": 1
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('storageAccountName')]",
+      "apiVersion": "2015-06-15",
+      "location": "[variables('location')]",
+      "properties": {
+        "accountType": "[variables('storageAccountType')]"
+      }
+    },
+    {
+      "apiVersion": "2015-06-15",
+      "type": "Microsoft.Network/virtualNetworks",
+      "name": "[variables('virtualNetworkName')]",
+      "location": "[variables('location')]",
+      "properties": {
+        "addressSpace": {
+          "addressPrefixes": [
+            "[variables('addressPrefix')]"
+          ]
+        },
+        "subnets": [
+          {
+            "name": "[variables('subnetName')]",
+            "properties": {
+              "addressPrefix": "[variables('subnetPrefix')]"
+            }
+          }
+        ]
+      }
+    },
+    {
+      "apiVersion": "2015-06-15",
+      "type": "Microsoft.Network/publicIPAddresses",
+      "copy": {
+        "name": "vmCountCopy",
+        "count": "[variables('VMCount')]"
+      },
+      "name": "[concat(variables('publicIPAddressName'),copyIndex(variables('VMStart')))]",
+      "location": "[variables('location')]",
+      "properties": {
+        "publicIPAllocationMethod": "[variables('publicIPAddressType')]",
+        "dnsSettings": {
+          "domainNameLabel": "[concat(variables('prefix'),'vm',copyIndex(variables('VMStart')))]"
+        }
+      }
+    },
+    {
+      "apiVersion": "2015-06-15",
+      "type": "Microsoft.Network/networkInterfaces",
+      "copy": {
+        "name": "vmCountCopy",
+        "count": "[variables('VMCount')]"
+      },
+      "name": "[concat(variables('nicName'),copyIndex(variables('VMStart')))]",
+      "location": "[variables('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.Network/publicIPAddresses/', variables('publicIPAddressName'),copyIndex(variables('VMStart')))]",
+        "[concat('Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'))]"
+      ],
+      "properties": {
+        "ipConfigurations": [
+          {
+            "name": "ipconfig1",
+            "properties": {
+              "privateIPAllocationMethod": "Dynamic",
+              "publicIPAddress": {
+                "id": "[resourceId('Microsoft.Network/publicIPAddresses',concat(variables('publicIPAddressName'),copyIndex(variables('VMStart'))))]"
+              },
+              "subnet": {
+                "id": "[variables('subnetRef')]"
+              }
+            }
+          }
+        ]
+      }
+    },
+    {
+      "apiVersion": "2015-06-15",
+      "type": "Microsoft.Compute/virtualMachines",
```

Comment:
This template aims at demonstrating using 'Zabbix' solution for monitoring a VM cluster.
Here we're just creating some VMs to emulate a user's VM cluster (as the monitoring target), and those VMs are not intend to fulfill certain tasks. So looks like using copy field here matches our scenario better.